core/block_svg.js

```diff
@@ -1374,15 +1326,40 @@ Blockly.BlockSvg.prototype.setDeleteStyle = function(enable) {
 // Overrides of functions on Blockly.Block that take into account whether the
 // block has been rendered.
 
+/**
+ * Get the colour of a block.
+ * @return {string} #RRGGBB string.
+ */
+Blockly.BlockSvg.prototype.getColour = function() {
+  return this.pathObject.primaryColour;
+};
```

Comment:
I would also add ``getColourSecondary`` and ``getColourTertiary`` wrapper methods.

Reply:
Added back.